cmd2/parsing.py

```diff
@@ -0,0 +1,387 @@
+            # everything before the first terminator is the command and the args
+            (command, args) = self._command_and_args(tokens[:terminator_pos])
+            # we will set the suffix later
+            # remove all the tokens before and including the terminator
+            tokens = tokens[terminator_pos+1:]
+        else:
+            (testcommand, testargs) = self._command_and_args(tokens)
+            if testcommand in self.multilineCommands:
+                # no terminator on this line but we have a multiline command
+                # everything else on the line is part of the args
+                # because redirectors can only be after a terminator
+                command = testcommand
+                args = testargs
+                tokens = []
+
+        # check for input from file
+        inputFrom = None
+        try:
+            input_pos = tokens.index('<')
+            inputFrom = ' '.join(tokens[input_pos+1:])
+            tokens = tokens[:input_pos]
+        except ValueError:
+            pass
+
+        # check for output redirect
+        output = None
+        outputTo = None
+        try:
+            output_pos = tokens.index('>')
+            output = '>'
+            outputTo = ' '.join(tokens[output_pos+1:])
+            # remove all the tokens after the output redirect
+            tokens = tokens[:output_pos]
+        except ValueError:
+            pass
+
+        try:
+            output_pos = tokens.index('>>')
+            output = '>>'
+            outputTo = ' '.join(tokens[output_pos+1:])
+            # remove all tokens after the output redirect
+            tokens = tokens[:output_pos]
+        except ValueError:
+            pass
+
+        # check for pipes
+        try:
+            # find the first pipe if it exists
+            pipe_pos = tokens.index('|')
+            # set everything after the first pipe to result.pipeTo
+            pipeTo = ' '.join(tokens[pipe_pos+1:])
+            # remove all the tokens after the pipe
+            tokens = tokens[:pipe_pos]
+        except ValueError:
+            # no pipe in the tokens
+            pipeTo = None
+        
+        if terminator:
+            # whatever is left is the suffix
+            suffix = ' '.join(tokens)
+        else:
+            # no terminator, so whatever is left is the command and the args
+            suffix = None
+            if not command:
+                # command could already have been set, if so, don't set it again
+                (command, args) = self._command_and_args(tokens)
+
+        # set multiline
+        if command in self.multilineCommands:
+            multilineCommand = command
+        else:
+            multilineCommand = None
+
+        # build Statement object
+        result = Statement(args)
+        result.raw = rawinput
+        result.command = command
+        result.args = args
+        result.terminator = terminator
+        result.inputFrom = inputFrom
```

Comment:
**inputFrom** member variable is not defined in the ``Statement`` class ``__init__()``